plexpy/notifiers.py

```diff
@@ -1574,3 +1614,202 @@ def return_config_options(self):
                          ]
 
         return config_option
+
+
+class Scripts(object):
+
+    def __init__(self, **kwargs):
+        pass
+
+    def conf(self, options):
+            return cherrypy.config['config'].get('Scripts', options)
+
+    def updateLibrary(self):
+        # For uniformity reasons not removed
+        return
+
+    def test(self, subject, message, *args, **kwargs):
+        self.notify(subject, message, *args, **kwargs)
+        return
+
+    def list_scripts(self):
+        scriptdir = plexpy.CONFIG.SCRIPTS_FOLDER
+        scripts = {}
+
+        if scriptdir and not os.path.exists(scriptdir):
+            os.makedirs(scriptdir)
+
+        for root, dirs, files in os.walk(scriptdir):
+            for f in files:
+                name, ext = os.path.splitext(f)
+                ext = ext[1:]
+                if ext in ('rb', 'pl', 'bat', 'py', 'sh', 'cmd', 'php'):
+                    fp = os.path.join(scriptdir, f)
+                    scripts[fp] = fp
+
+        return scripts
+
+    def notify(self, subject, message, notify_action=None, script_args=''):
+        logger.debug('Ran notify script subject: %s message: %s, action: %s script_args: %s' % (subject, message, notify_action, script_args))
+
+        prefix = ''
+        script = ''
+
+        if not plexpy.CONFIG.SCRIPTS_FOLDER:
+            return
+
+        # Make sure we use the correct script..
+        if notify_action == 'play':
+            script = plexpy.CONFIG.SCRIPTS_ON_PLAY_SCRIPT
+
+        elif notify_action == 'stop':
+            script = plexpy.CONFIG.SCRIPTS_ON_STOP_SCRIPT
+
+        elif notify_action == 'pause':
+            script = plexpy.CONFIG.SCRIPTS_ON_PAUSE_SCRIPT
+
+        elif notify_action == 'resume':
+            script = plexpy.CONFIG.SCRIPTS_ON_RESUME_SCRIPT
+
+        elif notify_action == 'buffer':
+            script = plexpy.CONFIG.SCRIPTS_ON_BUFFER_SCRIPT
+
+        elif notify_action == 'extdown':
+            script = plexpy.CONFIG.SCRIPTS_ON_EXTDOWN_SCRIPT
+
+        elif notify_action == 'extup':
+            script = plexpy.CONFIG.SCRIPTS_ON_EXTUP_SCRIPT
+
+        elif notify_action == 'intdown':
+            script = plexpy.CONFIG.SCRIPTS_ON_INTDOWN_SCRIPT
+
+        elif notify_action == 'intup':
+            script = plexpy.CONFIG.SCRIPTS_ON_INTUP_SCRIPT
+
+        elif notify_action == 'created':
+            script = plexpy.CONFIG.SCRIPTS_ON_CREATED_SCRIPT
+
+        name, ext = os.path.splitext(script)
+
+        if ext == '.py':
+            prefix = 'python'
+        elif ext == '.php':
+            prefix = 'php'
+        elif ext == '.pl':
+            prefix = 'perl'
+        elif ext == '.rb':
+            prefix = 'ruby'
+
+        script = script.split()
+
+        if prefix:
+            script.insert(0, prefix)
+
+        if script_args:
+            script = script + script_args.split()
```

Comment:
Yeah, i think it would be way to much. The most important thing for a script is action. 
I havnt tested with space in the args, it might break stuff atleast on windows.